### PR TITLE
More SQLite

### PR DESCRIPTION
**Pull request by neilalexander**
This PR adds SQLite support to `appservice`, `mediaapi` and `publicroomsapi`. 

Note that there are a couple of caveats with the `publicroomsapi` - the default queries return 30 rows (as we apparently have to specify a limit of some kind) and we don't check aliases in the filters yet.

### CODE REVIEW

File: publicroomsapi/storage/sqlite3/public_rooms_table.go [hunk 1]
```diff
@@ -39,26 +39,15 @@ var editableAttributes = []string{
...
-	-- Topic of the room (empty string if none)
 	topic TEXT NOT NULL DEFAULT '',
-	-- Is the room world readable?
 	world_readable BOOLEAN NOT NULL DEFAULT false,
-	-- Can guest join the room?
 	guest_can_join BOOLEAN NOT NULL DEFAULT false,
-	-- URL of the room avatar (empty string if none)
 	avatar_url TEXT NOT NULL DEFAULT '',
-	-- Visibility of the room: true means the room is publicly visible, false
-	-- means the room is private
```
Comment (anoadragon453) [inline publicroomsapi/storage/sqlite3/public_rooms_table.go]: Might be nice to keep these comments?

File: publicroomsapi/storage/sqlite3/public_rooms_table.go [hunk 2]
```diff
@@ -71,33 +60,33 @@ const selectPublicRoomsSQL = "" +
...
 
 const selectPublicRoomsWithFilterSQL = "" +
 	"SELECT room_id, joined_members, aliases, canonical_alias, name, topic, world_readable, guest_can_join, avatar_url" +
 	" FROM publicroomsapi_public_rooms" +
 	" WHERE visibility = true" +
 	" AND (LOWER(name) LIKE LOWER($1)" +
-	" OR LOWER(topic) LIKE LOWER($1)" +
-	" OR LOWER(ARRAY_TO_STRING(aliases, ',')) LIKE LOWER($1))" +
+	" OR LOWER(topic) LIKE LOWER($1))" +
+	//" OR LOWER(ARRAY_TO_STRING(aliases, ',')) LIKE LOWER($1))" +
```
Comment (anoadragon453) [inline publicroomsapi/storage/sqlite3/public_rooms_table.go]: Do we want to keep this?

File: publicroomsapi/storage/sqlite3/public_rooms_table.go [hunk 3]
```diff
@@ -71,33 +60,33 @@ const selectPublicRoomsSQL = "" +
...
 
 const selectPublicRoomsWithLimitAndFilterSQL = "" +
 	"SELECT room_id, joined_members, aliases, canonical_alias, name, topic, world_readable, guest_can_join, avatar_url" +
 	" FROM publicroomsapi_public_rooms" +
 	" WHERE visibility = true" +
 	" AND (LOWER(name) LIKE LOWER($1)" +
-	" OR LOWER(topic) LIKE LOWER($1)" +
-	" OR LOWER(ARRAY_TO_STRING(aliases, ',')) LIKE LOWER($1))" +
+	" OR LOWER(topic) LIKE LOWER($1))" +
+	// " OR LOWER(ARRAY_TO_STRING(aliases, ',')) LIKE LOWER($1))" +
```
Comment (anoadragon453) [inline publicroomsapi/storage/sqlite3/public_rooms_table.go]: and this

File: publicroomsapi/storage/sqlite3/public_rooms_table.go (hunk 2 above)
Comment (neilalexander) [reply publicroomsapi/storage/sqlite3/public_rooms_table.go]: I suppose I should `TODO` this, as the query as it is won't work with SQLite.

File: publicroomsapi/storage/sqlite3/public_rooms_table.go (hunk 3 above)
Comment (neilalexander) [reply publicroomsapi/storage/sqlite3/public_rooms_table.go]: Same with this one - I'll `TODO` it.

File: publicroomsapi/storage/sqlite3/public_rooms_table.go (hunk 1 above)
Comment (neilalexander) [reply publicroomsapi/storage/sqlite3/public_rooms_table.go]: For now I don't think any of the SQLite queries have comments as it made them easier to read, but we can re-add those later.

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**